library/src/main/java/org/apache/apex/malhar/lib/state/managed/IncrementalCheckpointManager.java

```diff
@@ -133,8 +134,10 @@ protected void transferWindowFiles()
           Map<Long, Map<Slice, Bucket.BucketedValue>> buckets = savedWindows.remove(windowId);
 
           for (Map.Entry<Long, Map<Slice, Bucket.BucketedValue>> singleBucket : buckets.entrySet()) {
-            managedStateContext.getBucketsFileSystem().writeBucketData(windowId, singleBucket.getKey(),
-                singleBucket.getValue());
+            long bucketId = singleBucket.getKey();
+            if (bucketId > latestPurgedTimeBucket) {
```

Comment:
I found an issue in MovingBoundaryTimeBucketAssigner, where the lower bound was initialized with 0 instead of -1, causing purging to be triggered with 0, which is a valid bucket ID. I also made another change to avoid trigger purging unless necessary. @chandnisingh I also see that `StateTrackerTest` takes a long time to run (3 minutes), is that expected?